src/AssociatedContentAssignment.js

```diff
@@ -5,19 +5,25 @@ import {
   generateFriendlyStringFromSubmissionFormats,
   getAssignmentSettingsHref
 } from "./utils";
+import PreviewUnavailable from "./PreviewUnavailable";
 
 export default class AssociatedContentAssignment extends Component {
+  constructor(props) {
+    super(props);
+    this.settings = null;
+  }
+
   async componentDidMount() {
     const parser = new DOMParser();
     const settingsXml = await this.props.getTextByPath(
       getAssignmentSettingsHref(this.props.identifier)
     );
-    const settings = parser.parseFromString(settingsXml, "text/xml");
-    const pointsPossibleNode = settings.querySelector("points_possible");
+    this.settings = parser.parseFromString(settingsXml, "text/xml");
```

Comment:
Hm. Maybe instead of attaching settings to the instance you could just save `external_tool_external_identifier` in a variable called `externalTool` and save it via setState(). 

On line 29 you can see where we are doing something similar with `pointsPossible` and `submissionFormats`.